app/src/test/java/com/rkm/tasky/feature/fakes/EmailPatternValidatorFake.kt

```diff
@@ -0,0 +1,10 @@
+package com.rkm.tasky.feature.fakes
+
+import com.rkm.tasky.util.validator.abstraction.EmailPatternValidator
+
+class EmailPatternValidatorFake: EmailPatternValidator {
+    override fun isValidEmail(email: String): Boolean {
+        val emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"
+        return email.matches(emailRegex.toRegex())
+    }
+}
```

Comment:
For a fake you'd rather want to be able to pass in what it should return. Otherwise you could just use your real instance right away if that's not what you need. Test doubles like fakes are meant to replace instances for tests to isolate another class. For example testing your ViewModel in isolation, so you pass in a fake for the validator that always returns true, so you know that the test can't fail because of any logic in the fake and must therefore be an issue in the ViewModel.